package_exporter/src/generate_package.rs

```diff
@@ -0,0 +1,195 @@
+    let launch_directory_name = "launch".to_string();
+    let urdf_directory_name = "urdf".to_string();
+    let rviz_directory_name = "rviz".to_string();
+
+    // Create paths
+    let output_directory_path = std::path::Path::new(&output_directory);
+    let output_package_path = output_directory_path.join(&new_package_name);
+    let meshes_directory_path = output_package_path.join(&mesh_directory_name);
+    let launch_directory_path = output_package_path.join(&launch_directory_name);
+    let urdf_directory_path = output_package_path.join(&urdf_directory_name);
+    let rviz_directory_path = output_package_path.join(&rviz_directory_name);
+
+    // Create directories
+    if output_package_path.exists() {
+        std::fs::remove_dir_all(&output_directory_path)?;
+    }
+    for directory_path in [
+        &output_package_path,
+        &meshes_directory_path,
+        &launch_directory_path,
+        &urdf_directory_path,
+        &rviz_directory_path,
+    ] {
+        std::fs::create_dir_all(directory_path)?;
+    }
+
+    // Create the package
+    write_urdf_and_copy_mesh_files(&workcell, &mesh_directory_name, &meshes_directory_path, &new_package_name, &urdf_directory_path);
+    generate_templates(
+        package_context,
+        &output_package_path,
+        &launch_directory_path,
+        &rviz_directory_path,
+    )?;
+
+    Ok(())
+}
+
+fn write_urdf_and_copy_mesh_files(
+    workcell: &Workcell,
+    mesh_directory_name: &str,
+    meshes_directory_path: &std::path::Path,
+    new_package_name: &str,
+    urdf_directory_path: &std::path::Path,
+) -> Result<(), Box<dyn Error>> {
+    let asset_paths = get_mesh_paths(&workcell)?;
+    copy_files(&asset_paths, &meshes_directory_path)?;
+
+    let new_workcell = convert_to_package_paths(&workcell, new_package_name, mesh_directory_name)?;
+
+    let urdf_robot = new_workcell.to_urdf()?;
+    let urdf_file_path = urdf_directory_path.join("robot.urdf");
+    let urdf_string = urdf_rs::write_to_string(&urdf_robot)?;
+    std::fs::write(urdf_file_path, urdf_string)?;
+
+    Ok(())
+}
+
+fn convert_to_package_paths(workcell: &Workcell, package_name: &str, mesh_directory_name: &str) -> Result<Workcell, Box<dyn Error>> {
+    let mut workcell = workcell.clone();
+    workcell.visuals.iter_mut().chain(workcell.collisions.iter_mut()).try_for_each(|(id, visual)| {
+        if let Geometry::Mesh { source: asset_source, scale } = &mut visual.bundle.geometry {
+            let path = get_path_to_asset_file(asset_source)?;
+
+            let file_name = PathBuf::from(&path).file_name().ok_or(
+                IoError::new(IoErrorKind::InvalidInput, "Unable to get file name from path")
+            )?.to_str().ok_or(
+                IoError::new(IoErrorKind::InvalidInput, "Unable to convert file name to str")
+            )?.to_owned();
+
+            let package_path = format!("package://{}/{}/{}", package_name, mesh_directory_name, file_name);
```

Comment:
Since the asset path already contains `package://`, this instruction seems to prepend another `package://` string, paths look like this on my machine: `package://package://package_name/path`, hence rviz is not able to resolve them.
```suggestion
            let package_path = format!("{}/{}/{}", package_name, mesh_directory_name, file_name);
```

Reply:
Should be fixed in c822bd5.